### PR TITLE
Alvin' PR

### PR DESCRIPTION
**Pull request by HoiDam**

### CODE REVIEW

**Comment by michael-nammi** [comment]
Business requirement
1. finish 13 test cases and test can pass
2. the last test case named NRML but the input is only "R" and the instantiation of MarsRover may have problems. (multiple command)

Focus on Practice
1. at least 20 Git commits
2. commit test case first then commit related implementation
3. commit message format is type: detail message
4.  use meaningful names for variables, methods, classes
5. use java stream instead of for loop
6. no obvious duplicate code
7. No useless code
8. no long method
9. no more than 4 if/else branches

File: src/main/java/com/afs/tdd/MarsRover.java [hunk 1]
```diff
@@ -1,4 +1,109 @@
...
+        return locationY;
+    }
+
+    public String getDirection() {
+        return direction;
+    }
+
+    public void executeCommands(String commands) {
+        Arrays.asList(commands).stream()
+            .map(command -> {
```
Comment (michael-nammi) [inline src/main/java/com/afs/tdd/MarsRover.java:104]: forEach may be the alternative

File: src/test/java/com/afs/tdd/MarsRoverTest.java [hunk 2]
```diff
@@ -2,9 +2,165 @@
...
+        assertEquals(0,marsRover.getLocationY());
+        assertEquals("N",marsRover.getDirection());
+    }
+    @Test
+    void should_go_to_1_1_N_when_execute_commands_given_0_0_N_MRML() {
+        //given
+        MarsRover marsRover = new MarsRover(1,1,"N");
+        //when
 
+        marsRover.executeCommands("R");
```
Comment (michael-nammi) [inline src/test/java/com/afs/tdd/MarsRoverTest.java:160]: does not match the test description

File: src/test/java/com/afs/tdd/MarsRoverTest.java [hunk 3]
```diff
@@ -2,9 +2,165 @@
...
+        marsRover.executeCommand("R");
+        //then
+        assertEquals(0,marsRover.getLocationX());
+        assertEquals(0,marsRover.getLocationY());
+        assertEquals("N",marsRover.getDirection());
+    }
+    @Test
+    void should_go_to_1_1_N_when_execute_commands_given_0_0_N_MRML() {
+        //given
+        MarsRover marsRover = new MarsRover(1,1,"N");
```
Comment (michael-nammi) [inline src/test/java/com/afs/tdd/MarsRoverTest.java:157]: initialization does not match the description

**Comment by HoiDam** [comment]
Archive

image

**Pull request closed without merging**